docs/tactics.md

```diff
@@ -199,22 +199,22 @@ Unfold coercion-related definitions
 
 ### Instance cache tactics
 
-* `resetI`: Reset the instance cache. This allows any new instances
-  added to the context to be used in typeclass inference.
+* `resetI`: Reset the instance cache. This allows any instances
+  currently in the context to be used in typeclass inference.
 
 * `unfreezeI`: Unfreeze local instances, which allows us to revert
   instances in the context
 
-* `introI`/`introsI`: Like `intro`/`intros`, but uses the introduced variable
-  in typeclass inference.
+* `introI`/`introsI`: `intro`/`intros` followed by `resetI`. Like
+  `intro`/`intros`, but uses the introduced variable in typeclass inference.
 
-* `haveI`/`letI`: Used to add typeclasses to the context so that they can
-  be used in typeclass inference. The syntax is the same as
-  `have`/`letI`, but the proof-omitted version of `have` is not supported
-  (for this one must write `have : t, { <proof> }, resetI, <proof>`).
+* `haveI`/`letI`: `have`/`let` followed by `resetI`. Used to add typeclasses
+  to the context so that they can be used in typeclass inference. The syntax
+  is the same as `have`/`letI`, but the proof-omitted version of `have` is
```

Comment:
this should be `have` / `let` and not `have` / `letI`

Reply:
Also the syntax isn't quite the same. The following works for `have`, but not `haveI`
```lean
example : false := begin
  haveI : ring ℤ, from _
end
```

Reply:
It says "the proof-omitted version of `have` is not supported". Admittedly, the existence of the `from` tactic makes it a little non obvious that this is actually two tactics, since it resembles the proof term version that treats both parts as a unit.

Reply:
I wrote a little more to clarify these things.